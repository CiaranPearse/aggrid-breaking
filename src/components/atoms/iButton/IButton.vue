<template>
  <v-btn v-if="buttonText && buttonText !== '' && !buttonIcon" :variant="variant" :color="color" :disabled="disabled" :size="size"
    @click.stop="() => buttonPressed(intent, identifier, action)">
    {{ buttonText }}
  </v-btn>
  <v-btn v-else-if="buttonText && buttonText !== '' && buttonIcon" :variant="variant" :color="color" :disabled="disabled" :size="size"
    @click.stop="() => buttonPressed(intent, identifier, action)" :prepend-icon="buttonIcon">
    {{ buttonText }}
  </v-btn>
  <v-btn v-else :icon="buttonIcon" :color="color" :disabled="disabled" :size="size"
    @click.stop="() => buttonPressed(intent, identifier, action)"></v-btn>
</template>

<script setup lang="ts">
import { inject } from 'vue';
import './iButton.css';
import { Variants } from '../../../constants';
import { Colors } from '../../../constants/text/colors';
import router from '../../../router';
import { IButtonPressEvent } from '../../../types'

const emitter = inject('emitter');
const props = defineProps({
  buttonText: {
    type: String,
    required: false,
  },
  buttonIcon: {
    type: String,
    required: false,
  },
  variant: {
    type: String,
    required: false,
    default: Variants.Default,
  },
  size: {
    type: String,
    default: 'default',
  },
  color: {
    type: String,
    default: Colors.Primary,
  },
  disabled: {
    type: Boolean,
    default: false,
  },
  intent: {
    type: String,
    required: true,
    default: 'func',
  },
  identifier: {
    type: String,
  },
  action: {
    type: String,
  }
})


const buttonPressed = (intent: string, identifier: string | undefined, action?: string | undefined): void => {
  if (intent === 'link' && action) {
    router.push(action)
  } else {
    // TO DO - FIX THIS IGNORE
    //@ts-ignore
    emitter.emit(identifier, 100);
    return
  }
}

</script>
