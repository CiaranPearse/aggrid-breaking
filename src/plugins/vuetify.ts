import '@mdi/font/css/materialdesignicons.css';
import 'vuetify/styles';
import { createVuetify } from 'vuetify';
import * as components from 'vuetify/components';
import * as directives from 'vuetify/directives';

export const defaultLight = {
  dark: false,
  colors: {
    background: 'light-blue',
    layoutBackground: '#0071c5',
    surface: '#FFFFFF',
    primary: '#0071c5',
    'primary-darken-1': '#3700B3',
    secondary: '#03DAC6',
    'secondary-darken-1': '#018786',
    error: '#B00020',
    info: '#2196F3',
    success: '#4CAF50',
    warning: '#FB8C00',
  },
};
export const darkOne = {
  dark: true,
  colors: {
    background: '#FFFFFF',
    layoutBackground: '#98c3e3',
    surface: '#FFFFFF',
    primary: '#0071c5',
    'primary-darken-1': '#3700B3',
    secondary: '#03DAC6',
    'secondary-darken-1': '#018786',
    error: '#B00020',
    info: '#2196F3',
    success: '#4CAF50',
    warning: '#FB8C00',
  },
};

export default createVuetify({
  components,
  directives,
  theme: {
    defaultTheme: 'defaultLight',
    themes: {
      defaultLight,
      darkOne,
    },
  },
});
