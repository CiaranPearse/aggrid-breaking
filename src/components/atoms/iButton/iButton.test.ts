import { createVuetify } from "vuetify";
import { mount } from '@vue/test-utils';
import IButton from './IButton.vue';
import { useRouter, useRoute } from 'vue-router'
import jest from 'jest-mock'

const vuetify = createVuetify();



describe('Button', () => {
  
  it('should render', () => {
    const wrapper = mount(IButton, {
      props: {
        buttonText: "Test Button",
        color: 'primary',
        intent: 'link'
      },
      global: {
          plugins: [vuetify],
      },
  });
    expect(IButton).toBeTruthy()
  })

  it('should render as a Flat Button', () => {
    const flatButtonText = 'Flat Button';
    const wrapper = mount(IButton, {
      props: {
        buttonText: flatButtonText,
        variant: 'flat',
        color: 'primary',
        intent: 'link'
      },
      global: {
          plugins: [vuetify],
      },
  });
  expect(wrapper.text()).toMatch(flatButtonText);
  const correctClass = wrapper.find('.v-btn--variant-flat');
  expect(correctClass.exists()).toBe(true);
  expect(wrapper.html()).toMatchSnapshot();
  })

  it('should render as an Icon Button', () => {
    const wrapper = mount(IButton, {
      props: {
        buttonIcon: 'mdi-home-outline',
        buttonText: '',
        color: 'primary',
        intent: 'link'
      },
      global: {
          plugins: [vuetify],
      },
  });
  const correctClass = wrapper.find('.mdi-home-outline');
  expect(correctClass.exists()).toBe(true);
  expect(wrapper.html()).toMatchSnapshot();
  })

  it('should render as an Icon and text button', () => {
    const flatButtonText = 'Flat Button';
    const wrapper = mount(IButton, {
      props: {
        buttonText: flatButtonText,
        buttonIcon: 'mdi-home-outline',
        variant: 'flat',
        color: 'primary',
        intent: 'link'
      },
      global: {
          plugins: [vuetify],
      },
  });
  expect(wrapper.text()).toMatch(flatButtonText);
  const correctClass = wrapper.find('.mdi-home-outline');
  expect(correctClass.exists()).toBe(true);
  expect(wrapper.html()).toMatchSnapshot();
  })

  it('should emit a value when clicked if function', async () => {
    const flatButtonText = 'Flat Button';
    const wrapper = mount(IButton, {
      props: {
        buttonText: flatButtonText,
        buttonIcon: 'mdi-home-outline',
        variant: 'flat',
        color: 'primary',
        intent: 'func',
        identifier: 'testIdentifier',
        action: 'textAction'
      },
      global: {
          plugins: [vuetify],
      },
  });
  wrapper.find('button').trigger('click')
  wrapper.vm.$emit('testIdentifier', 'testAction')
  await wrapper.vm.$nextTick()
  expect(wrapper.emitted().testIdentifier).toBeTruthy();
  expect(wrapper.emitted().testIdentifier[0]).toEqual(['testAction'])
  })
})