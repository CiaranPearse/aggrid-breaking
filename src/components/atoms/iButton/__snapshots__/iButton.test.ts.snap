// Vitest Snapshot v1

exports[`Button > should render as a Flat Button 1`] = `
"<button type=\\"button\\" class=\\"v-btn v-theme--light bg-primary v-btn--density-default v-btn--size-default v-btn--variant-flat\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
  <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\">Flat Button</span>
  <!---->
  <!---->
</button>"
`;

exports[`Button > should render as an Icon Button 1`] = `
"<button type=\\"button\\" class=\\"v-btn v-btn--elevated v-btn--icon v-theme--light bg-primary v-btn--density-default v-btn--size-default v-btn--variant-elevated\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
  <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-home-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
  <!---->
  <!---->
</button>"
`;

exports[`Button > should render as an Icon and text button 1`] = `
"<button type=\\"button\\" class=\\"v-btn v-theme--light bg-primary v-btn--density-default v-btn--size-default v-btn--variant-flat\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><span class=\\"v-btn__prepend\\"><i class=\\"mdi-home-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span><span class=\\"v-btn__content\\" data-no-activator=\\"\\">Flat Button</span>
  <!---->
  <!---->
</button>"
`;
